gcloud-java-storage/src/main/java/com/google/gcloud/storage/Storage.java

```diff
@@ -501,6 +502,14 @@ public static Builder builder() {
   BlobInfo create(BlobInfo blobInfo, byte[] content, BlobTargetOption... options);
 
   /**
+   * Create a new blob.
+   *
+   * @return a complete blob information.
+   * @throws StorageException upon failure
+   */
+  BlobInfo createFromStream(BlobInfo blobInfo, InputStream content, BlobTargetOption... options);
```

Comment:
I think that `null` was only supported to allow the creation of a blob with no content. What do we do then? I'm ok with renaming `createFromStream` to `create`, should I add a `datastore.create(blobInfo)` as well?